src/ex2/app.js

```diff
@@ -0,0 +1,124 @@
+import { ItemManager } from "./itemManager.js"
+import { PokemonClient } from "./pokemonClient.js"
+
+const addButton = document.getElementById("list-item-submit")
+const textInput = document.getElementById("list-item-input")
+const allLists = document.getElementById("list")
+const counterLists = document.getElementById("counter")
+const deleteAllLists = document.getElementById("deleteAllTasks")
+const errorMessage = document.getElementById("error")
+const regex = /^[0-9]*$/
+
+const itemManager = new ItemManager()
+const pokemonClient = new PokemonClient()
+class Main {
+  constructor() {
+    this.name = "Main"
+  }
+  init() {
+    addButton.addEventListener("click", function () {
+      validation()
+    })
+  }
+}
+
+function addNewList(name) {
+  const li = document.createElement("li")
+  const hr = document.createElement("hr")
+  const deleteButton = document.createElement("button")
+  deleteButton.setAttribute("class", "dltbtn")
+  deleteButton.innerHTML = "X"
+
+  deleteButton.addEventListener("click", function () {
+    li.remove()
+    hr.remove()
+    itemManager.removeItem(textInput.value)
+    countLists()
+  })
+  li.addEventListener("mouseover", function () {
+    deleteButton.style.display = "block"
+    deleteButton.style.transition = "all 0.5s ease-in-out"
+  })
+  li.addEventListener("mouseout", function () {
+    deleteButton.style.display = "none"
+    deleteButton.style.transition = "all 0.5s ease-in-out"
+  })
+  li.appendChild(document.createTextNode(name))
+  li.appendChild(deleteButton)
+  allLists.appendChild(li)
+  allLists.appendChild(hr)
+  textInput.value = ""
+}
+
+async function validation() {
```

Comment:
this is a big function with a lot of code duplication, try to separate into smaller functions with single responsibility, so you could re-use them, and it will be easier to read